modules/core/src/main/scala/org/scalasteward/core/nurture/NurtureAlg.scala

```diff
@@ -77,37 +80,43 @@ final class NurtureAlg[F[_]](
       grouped = Update.group(filtered)
       _ <- logger.info(util.logger.showUpdates(grouped))
       baseSha1 <- gitAlg.latestSha1(repo, baseBranch)
-      _ <- grouped.traverse_ { update =>
+      result <- grouped.traverse { update =>
         val data =
           UpdateData(repo, fork, repoConfig, update, baseBranch, baseSha1, git.branchFor(update))
         processUpdate(data)
       }
-    } yield ()
+    } yield toFoldableOps(result).fold
 
   def getNonSbtUpdates(repo: Repo): F[List[Update.Single]] =
     for {
       maybeScalafmt <- scalafmtAlg.getScalafmtUpdate(repo)
     } yield List(maybeScalafmt).flatten
 
-  def processUpdate(data: UpdateData): F[Unit] =
+  def processUpdate(data: UpdateData): F[Either[Throwable, Unit]] =
     for {
       _ <- logger.info(s"Process update ${data.update.show}")
       head = vcs.listingBranch(config.vcsType, data.fork, data.update)
       pullRequests <- vcsApiAlg.listPullRequests(data.repo, head, data.baseBranch)
-      _ <- pullRequests.headOption match {
+      result <- pullRequests.headOption match {
         case Some(pr) if pr.isClosed =>
-          logger.info(s"PR ${pr.html_url} is closed")
+          logger.info(s"PR ${pr.html_url} is closed").asRight
         case Some(pr) if data.repoConfig.updatePullRequests =>
-          logger.info(s"Found PR ${pr.html_url}") >> updatePullRequest(data)
+          logger.info(s"Found PR ${pr.html_url}") >>
+            updatePullRequest(data).attempt
```

Comment:
One potential problem I see here is that `updatePullRequest` may leave the working copy of the repository in a dirty state which may affect and corrupt the next `processUpdate` call. Suppose `updatePullRequest` leaves changed files around which will be committed during the next `processUpdate` call. I guess the only way to prevent problems caused by a dirty state is to use a fresh clone after `updatePullRequest` failed. But I would rather try to fix `updatePullRequest` instead of making it really fail safe.

Reply:
A lighter approach would be to do a hard reset in case of a failure and let it throw in case it fails

Reply:
I would first like to understand why you want to make `updatePullRequest` fail safe at all. In which cases does `updatePullRequest` fail and why can't we fix these problems first?

Reply:
As far as I remember the problems were not inherent to scala-steward. Once there was an IOException because of a network-glitch or similar, another time a user deleted the origin/update/Xy branch and this caused a problem and there were other user related problems. My first priority is actually not the fail safe aspect but that the process as a whole fails if a problem occurs during the update process. For me it is fine if I change the PR in this direction. The fail-safe aspect is IMO nice as it works around the above mentioned problems